jena-arq/src/main/java/org/apache/jena/sparql/engine/http/QueryEngineHTTP.java

```diff
@@ -538,13 +539,29 @@ else if ( actualContentType.equals(WebContent.contentTypeJSON))
     @Override
     public JsonArray execJson()
     {
-        throw new NotImplemented("JSON queries not implemented for remote calls") ;
+        checkNotClosed();
+        HttpQuery httpQuery = makeHttpQuery();
+        httpQuery.setAccept(WebContent.contentTypeJSON);
+        InputStream in = httpQuery.exec();
+        JsonValue v = JSON.parseAny(in);
+        if ( ! v.isArray() )
+        {}
+        return v.getAsArray();
     }
 
     @Override
     public Iterator<JsonObject> execJsonItems()
     {
-        throw new NotImplemented("JSON queries not implemented for remote calls") ;
+        // Non-streaming.
+        // TODO Integrate with the JSON parser to stream the results. 
+        JsonArray array = execJson().getAsArray();
+        List<JsonObject> x = new ArrayList<>(array.size());
+        array.forEach(elt->{
+            if ( ! elt.isObject()) 
```

Comment:
Are these two `if` necessary?

Reply:
They are checking expectations to be defensive against misbehaving remote servers. Otherwise the app will get a less helpful exception on the attempt cast performed by `getAsArray` or `getAsObject`.

Reply:
Oh, now I see the `throew new QueryExecException`, but I could swear I had seen as `if (...) {}` before in GitHub. My bad.

Reply:
Another way:
```
return array.stream().peek(e -> if (!elt.isObject()) throw etc. )
                     .map(JsonValue::getAsObject)
                     .iterator();
```

Reply:
You did see `If ( ) { }` :-) I amended the commit and forced the push to my repo so the changes as history in master will be the changes all together in a single commit.